pkg/model/estimate.go

```diff
@@ -0,0 +1,99 @@
+package model
+
+import (
+    "os/exec"
+	"net"
+	"encoding/json"
+	"log"
+	// "os"
+)
+
+const (
+	SERVE_SOCKET = "/tmp/estimator.sock"
+)
+
+type Estimator struct {
+	Err chan error	
+	*exec.Cmd
+}
+
+type PowerRequest struct {
+	ModelName string   `json:"model_name"`
+	XCols 	[]string 	`json:"x_cols"`
+	XValues [][]float32 `json:"x_values"`
+	CorePower []float32 `json:"core_power"`
+	DRAMPower []float32 `json:"dram_power"`
+	GPUPower []float32 `json:"gpu_power"`
+	OtherPower []float32 `json:"other_power"`
+}
+
+type PowerResponse struct {
+	Powers []float32   `json:"powers"`
+	Message string  	`json:"msg"`
+}
+
+func (e *Estimator) StartPyEstimator() {
+    e.Cmd = exec.Command("python3", "./py/estimator.py")
+	go func() {
+		e.Err <- e.Cmd.Start()
+	}()
+	select{
+		case err := <- e.Err:
+			log.Printf("error: %v", err)
+	}
+}
+
+func (e *Estimator) Destroy() {
+	if err := e.Cmd.Process.Kill(); err != nil {
+		log.Printf("failed to kill process: %v", err)
+	} else {
+		log.Printf("kill estimator")
+	}
+}
+
+
+func (e *Estimator) GetPower(modelName string, xCols []string, xValues [][]float32, corePower, dramPower, gpuPower, otherPower []float32) []float32 {
```

Comment:
Does `GetPower` get metrics from all Pods as input, and get power consumption in a batch?

Reply:
Power consumption will be reported as a list of RAPL power package while xCols, xValues refers to read metric values such as cpu_cycles per pod (for all pods at the same time).

For example, 
at ticker t, 
RAPL pkg0 - core = 20, dram = 5, gpu = unknown, other = unknown 
RAPL pkg1 - core =30, dram = 1, gpu = unknown, other = unknown 

There are 3 pods
Pod A: cpu_cycles = 100, cache_miss=1
Pod B: cpu_cycles = 50, cache_miss=0
Pod C: cpu_cycles = 10, cache_miss=0

xCols = [cpu_cycles, cache_miss]
xValues = [[100, 1], [50, 0], [10, 0]]
corePower = [20, 30]
dramPower = [5, 1]
gpuPower = []
otherPower = []